stdlib/http/src/main/java/org/ballerinalang/net/http/nativeimpl/connection/ResponseWriter.java

```diff
@@ -0,0 +1,186 @@
+     */
+    public static void sendResponseRobust(DataContext dataContext, HttpCarbonMessage requestMessage, BMap<String,
+            BValue> outboundResponse, HttpCarbonMessage responseMessage) {
+        String contentType = HttpUtil.getContentTypeFromTransportMessage(responseMessage);
+        String boundaryString = null;
+        if (HeaderUtil.isMultipart(contentType)) {
+            boundaryString = HttpUtil.addBoundaryIfNotExist(responseMessage, contentType);
+        }
+
+        HttpMessageDataStreamer outboundMsgDataStreamer = getResponseDataStreamer(responseMessage);
+        HttpResponseFuture outboundRespStatusFuture = HttpUtil.sendOutboundResponse(requestMessage, responseMessage);
+        HttpConnectorListener outboundResStatusConnectorListener =
+                new ResponseWriter.HttpResponseConnectorListener(dataContext, outboundMsgDataStreamer);
+        outboundRespStatusFuture.setHttpConnectorListener(outboundResStatusConnectorListener);
+
+        OutputStream messageOutputStream = outboundMsgDataStreamer.getOutputStream();
+        BMap<String, BValue> entityStruct = extractEntity(outboundResponse);
+        if (entityStruct != null) {
+            if (boundaryString != null) {
+                serializeMultiparts(boundaryString, entityStruct, messageOutputStream);
+            } else {
+                BValue outboundMessageSource = EntityBodyHandler.getMessageDataSource(entityStruct);
+                serializeDataSource(outboundMessageSource, entityStruct, messageOutputStream);
+            }
+        }
+    }
+
+    /**
+     * Serialize multipart entity body. If an array of body parts exist, encode body parts else serialize body content
+     * if it exist as a byte channel.
+     *
+     * @param boundaryString      Boundary string that should be used in encoding body parts
+     * @param entity              Represents the entity that holds the actual body
+     * @param messageOutputStream Represents the output stream
+     */
+    private static void serializeMultiparts(String boundaryString, BMap<String, BValue> entity,
+                                            OutputStream messageOutputStream) {
+        BRefValueArray bodyParts = EntityBodyHandler.getBodyPartArray(entity);
+        try {
+            if (bodyParts != null && bodyParts.size() > 0) {
+                MultipartDataSource multipartDataSource = new MultipartDataSource(entity, boundaryString);
+                serializeDataSource(multipartDataSource, entity, messageOutputStream);
+                HttpUtil.closeMessageOutputStream(messageOutputStream);
+            } else {
+
+                EntityBodyHandler.writeByteChannelToOutputStream(entity, messageOutputStream);
+                HttpUtil.closeMessageOutputStream(messageOutputStream);
+            }
+        } catch (IOException e) {
+            throw new BallerinaException("Error occurred while serializing byte channel content : " +
```

Comment:
Shall we pass the exception object as a second parameter for the BallerinaException?